electrumsv/tests/test_wallet.py

```diff
@@ -550,64 +619,169 @@ def test_legacy_wallet_loading(mock_wallet_app_state, storage_info: WalletStorag
     with unittest.mock.patch(
             'electrumsv.keystore.app_state.credentials.add_indefinite_credential',
             add_indefinite_credential_mock):
-        try:
-            wallet = Wallet(storage)
-        except FileNotFoundError as e:
-            if sys.version_info[:3] >= (3, 8, 0):
-                msg = "Could not find module 'libusb-1.0.dll' (or one of its dependencies)."
-                if msg in e.args[0]:
-                    pytest.xfail("libusb DLL could not be found")
-                    return
-            raise e
-        except OSError as e:
-            if sys.version_info[:3] < (3, 8, 0):
-                if "The specified module could not be found" in e.args[1]:
-                    pytest.xfail("libusb DLL could not be found")
-                    return
-            raise e
-
-    # Store any pre-password update related data to compare against post-password update data.
-    prv_keypairs: dict[str, str] = {}
+        wallet = load_wallet(storage)
+        assert wallet is not None
+
+    # ---- Capture all types of encrypted data BEFORE the password update --- #
+
+    # Password token (probably an unnecessary check but it keeps a systematic approach)
+    password_token_before = wallet.get_storage().get('password-token')
+    if password_token_before is not None:
+        test_state.count_of_password_token_reencryptions += 1
+
+    # Derivation data in the KeyInstances table. Only for ImportedPrivkeyAccount types
+    keyinstance_private_keys: dict[str, str] = {}
     if "imported" == expected_type and "privkey" in wallet_filename:
         assert len(wallet.get_accounts()) == 2
         private_key_account = cast(ImportedPrivkeyAccount,
             [ entry for entry in wallet.get_accounts() if not entry.is_petty_cash() ][0])
         private_key_keystore = cast(Imported_KeyStore, private_key_account.get_keystore())
         # Pre-decrypt the prv for later comparison so the initial password is not needed there.
         for public_key, encrypted_prv in private_key_keystore._keypairs.items():
-            prv_keypairs[public_key.to_hex()] = pw_decode(encrypted_prv, initial_password)
-
-    password = "654321"
-    future, update_completion_event = wallet.update_password(initial_password, password)
+            keyinstance_private_keys[public_key.to_hex()] = pw_decode(encrypted_prv,
+                initial_password)
+            test_state.count_of_keyinstance_privkey_reencryptions += 1
```

Comment:
The whole point of counting is that we are counting that we have checked that we have successfully re-encrypted, increasing the count here I do not understand what it means.